Library/Homebrew/test/rubocops/lines_spec.rb

```diff
@@ -58,3 +58,75 @@ class Foo < Formula
     end
   end
 end
+
+describe RuboCop::Cop::FormulaAudit::GenerateCompletionsDSL do
+  subject(:cop) { described_class.new }
+
+  it "reports an offense when writing to a shell completions file directly" do
+    expect_offense(<<~RUBY)
+      class Foo < Formula
+        name "foo"
+
+        def install
+          (bash_completion/"foo").write Utils.safe_popen_read(bin/"foo", "completions", "bash")
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `generate_completions_from_executable(bin/"foo", "completions", shells: [:bash])` instead of `(bash_completion/"foo").write Utils.safe_popen_read(bin/"foo", "completions", "bash")`.
+        end
```

Comment:
This test fails at the moment, as the RuboCop expects an offense including `base_name` in the correction.
It seems like `@formula_name` is not instantiated correctly, since we are loading the formula from a heredoc and not a file path. Therefore, the `unless base_name == @formula_name` assertion in the Cop fails, and includes the `base_name` in the replacement.
Any suggestions how to solve this? I tried setting `cop.file_path= "foo.fb", but this did not help.
Should I include the `base_name` in the expected offense to make the test pass, even if this is not the wanted behavior?

Reply:
Try changing the `expect_offense` line to:

```ruby
expect_offense(<<~RUBY, "/homebrew-core/Formula/foo.rb")
  class Foo < Formula
  ...
RUBY
```